schema/bom-1.7.proto

```diff
@@ -2336,22 +2484,68 @@ message CryptoProperties {
       repeated string algorithms = 2;
       // A list of common identifiers for the cipher suite. For example: 0xC0 and 0x9E
       repeated string identifiers = 3;
+      // A list of TLS named groups (formerly known as curves) for this cipher suite. These groups define the parameters for key exchange algorithms like ECDHE.
+      repeated string tlsGroups = 4;
+      // A list of signature schemes supported for cipher suite. These schemes specify the algorithms used for digital signatures in TLS handshakes and certificate verification.
+      repeated string tlsSignatureSchemes = 5;
+    }
+
+    // Encryption Algorithm (ENCR)
+    message IkeV2Enc {
+      // A name for the encryption method
+      optional string name = 1;
+      // The key length of the encryption algorithm
+      optional int32 keyLength = 2;
+      // The bom-ref to algorithm cryptographic asset
+      optional string algorithm = 3;
+    }
+
+    // Pseudorandom Function (PRF)
+    message IkeV2Prf {
+      // A name for the pseudorandom function
+      optional string name = 1;
+      // The bom-ref to algorithm cryptographic asset
+      optional string algorithm = 2;
+    }
+
+    // Integrity Algorithm (INTEG)
+    message IkeV2Integ {
+      // A name for the integrity algorithm
+      optional string name = 1;
+      // The bom-ref to algorithm cryptographic asset
+      optional string algorithm = 2;
+    }
+
+    // Key Exchange Method (KE)
+    message IkeV2Ke {
+      // A group identifier for the key exchange algorithm
+      optional int32 group = 1;
+      // The bom-ref to algorithm cryptographic asset
+      optional string algorithm = 2;
+    }
+
+    // IKEv2 Authentication method
+    message IkeV2Auth {
+      // A name for the authentication method
+      optional string name = 1;
+      // The bom-ref to algorithm cryptographic asset
+      optional string algorithm = 2;
     }
 
     // IKEv2 Transform Types
     message Ikev2TransformTypes {
       // Transform Type 1: encryption algorithms
-      repeated string encr = 1;
+      repeated IkeV2Enc encr = 1;
       // Transform Type 2: pseudorandom functions
-      repeated string prf = 2;
```

Comment:
was a string, became a message.
:x: this is a breaking change.

Reply:
Same.